sdk/transaction.go

```diff
@@ -36,6 +32,106 @@ func (txs *TransactionService) GetTransaction(ctx context.Context, id string) (T
 	return MapTransaction(&b, txs.client.GenerationHash())
 }
 
+// returns an array of Transaction's for passed array of transaction ids or hashes
+func (txs *TransactionService) getTransactionsByGroup(ctx context.Context, group TransactionGroups, tpOpts *TransactionsPageOptions) (*TransactionsPage, error) {
+	tspDTO := &transactionsPageDTO{}
+
+	u, err := addOptions(fmt.Sprintf(transactionsByGroupRoute, group.String()), tpOpts)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := txs.client.doNewRequest(ctx, http.MethodGet, u, nil, &tspDTO)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = handleResponseStatusCode(resp, map[int]error{404: ErrResourceNotFound, 409: ErrArgumentNotValid}); err != nil {
+		return nil, err
+	}
+
+	return tspDTO.toStruct(txs.client.GenerationHash())
+}
+
+// returns an array of Transaction's for passed array of transaction ids or hashes
+func (txs *TransactionService) getTransactionsByIds(ctx context.Context, group TransactionGroups, ids []string, tpOpts *TransactionsPageOptions) ([]Transaction, error) {
+	var b bytes.Buffer
+	txIds := &TransactionIdsDTO{
+		ids,
+	}
+
+	u, err := addOptions(fmt.Sprintf(transactionsByGroupRoute, group.String()), tpOpts)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := txs.client.doNewRequest(ctx, http.MethodPost, u, txIds, &b)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = handleResponseStatusCode(resp, map[int]error{404: ErrResourceNotFound, 409: ErrArgumentNotValid}); err != nil {
+		return nil, err
+	}
+
+	return MapTransactions(&b, txs.client.GenerationHash())
+}
+
+// returns Transaction for passed transaction id or hash
+func (txs *TransactionService) GetAnyTransactionById(ctx context.Context, id string) (Transaction, error) {
+	trS, err := txs.GetTransactionStatus(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+
+	return txs.getTransaction(ctx, trS.Group, id)
+}
```

Comment:
@xgreenx, is it ok to do two requests to get any transactions??? Seems like a design flaw

Reply:
@xgreenx, I think we need it in SDK, but the question is: Why do we need the status of a transaction while we already have its Hash/ID?